src/Modules/Main.lua

```diff
@@ -717,6 +727,37 @@ function main:SaveSettings()
 	end
 end
 
+function main:OpenPathPopup(invalidPath)
+	local controls = { }
+	local defaultLabelPlacementX = 8
+
+	controls.label = new("LabelControl", { "TOPLEFT", nil, "TOPLEFT" }, defaultLabelPlacementX, 20, 206, 16, function()
+		return "^7User settings path contains unicode characters and cannot be loaded."..
+		"\nCurrent Path: "..invalidPath:gsub("?", "^1?^7").."/Path of Building/"..
+		"\nSpecify a new location for your Settings.xml:"
+	end)
+	controls.userPath = new("EditControl", { "TOPLEFT", controls.label, "TOPLEFT" }, 0, 60, 206, 20, invalidPath, nil, nil, nil, function(buf)
+		invalidPath = sanitiseText(buf)
+		if not invalidPath:match("?") then
+			controls.save.enabled = true
+		else
+			controls.save.enabled = false
+		end
+	end)
+	controls.save = new("ButtonControl", { "TOPLEFT", controls.userPath, "TOPLEFT" }, 0, 26, 206, 20, "Save", function()
+		local res, msg = MakeDir(controls.userPath.buf)
+		if not res then
+			main:OpenMessagePopup("Error", "Couldn't create '"..controls.userPath.buf.."' : "..msg)
+		end
+		self.userPath = controls.userPath.buf
```

Comment:
Does this end well if the user enters a relative path?

Reply:
It ends up using the directory that contains `Launch.lua` at the root in that case.  I was hoping for a better way to test for a valid path, but `MakeDir` returns `nil` even when the path already exists, and doesn't prepend the full path, so it's hard to give useful feedback to the user in those cases (especially when their `Launch.lua` exists in the User directory also)